src/Expander/Task.hs

```diff
@@ -36,18 +37,18 @@ data ExpanderTask
   | AwaitingDefn Var Ident Binding SplitCorePtr Ty SplitCorePtr Syntax
     -- ^ Waiting on var, binding, and definiens, destination, syntax to expand
   | AwaitingType SplitTypePtr [AfterTypeTask]
-  | ExpandDecl DeclPtr Scope Syntax DeclValidityPtr
-    -- ^ Where to put it, the scope, the decl, where to put the phase
-  | ExpandMoreDecls ModBodyPtr Scope Syntax DeclValidityPtr
-    -- Depends on the binding of the name(s) from the decl and the phase
+  | ExpandDeclForm DeclTreePtr DeclValidityPtr DeclValidityPtr Syntax
+    -- ^ The tree node to fill, the input environment, the output environment, and the declaration form.
+  | ExpandDeclForms DeclTreePtr DeclValidityPtr DeclValidityPtr Syntax
+    -- ^ The tree node to fill, the input environment, the output environment, and the declaration forms.
```

Comment:
There used to be two types of tasks, there would now need to be 2*2=4, depending on whether the input syntax is a single declaration form or a list of declaration forms, and on whether the input environment is a `DeclValidityPtr` we're waiting for or a `ScopeSet` we already have. For simplicity, I now call `expandDeclForm[s]` directly instead of forking a task when I already have a `ScopeSet`.